File: cpp/apps/meta/Exit.cpp
#include "Exit.hpp"
#include <iostream>
#include <cstdlib>
using namespace std;

void meta::finalize(const string &msg, meta::Status status)
{
    int ret = 0;
    switch (status)
    {
        case meta::Status::Ok: break;
        case meta::Status::Error:
                               cout << "ERROR::";
                               ret = -1;
                               break;
    }
    cout << msg << endl;
    exit(ret);
}

File: ruby/gubg/exit.rb
def finalize(msg, code = :error)
	prefix = (code == :error ? "ERROR::" : "")
	puts("#{prefix}#{msg}")
	exit(
		case code
		when :ok
			0
		when :error
			-1
		else
			-2
		end
	)
end
